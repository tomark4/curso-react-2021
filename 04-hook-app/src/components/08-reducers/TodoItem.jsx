const TodoItem = ({todo, onDelete, onTouch}) => {
    return (
        <li className="list-group-item">
            <div className="row">
                <div className="col-md-8">
                    <p className={todo.done ? 'completed' : ''}
                    onClick={() => onTouch(todo.id)}>{todo.name}</p>
                </div>
                <div className="col-md-4 text-right">
                    <button className="btn btn-danger btn-sm"
                    onClick={() => onDelete(todo.id)}>Borrar</button>
                </div>
            </div>
        </li>
    )
}

export default TodoItem;
