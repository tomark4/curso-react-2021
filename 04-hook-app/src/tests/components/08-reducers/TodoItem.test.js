import { shallow } from "enzyme";
import TodoItem from "../../../components/08-reducers/TodoItem";
import {demoTodos} from '../../fixtures/demoTodo';

describe('pruebas en todo item', () => {
  const handleDelete = jest.fn();
  const handleTouch = jest.fn();

  const wrapper = shallow(<TodoItem
    todo={ demoTodos[0] }
    onDelete={ handleDelete }
    onTouch={ handleTouch }/>);

  test('should mostrar el component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should llamar la funcion borrar', () => {
    wrapper.find('.btn-danger').simulate('click');
    expect(handleDelete).toHaveBeenCalledWith( demoTodos[0].id );
  });

  test('should llamar la funcion toggle', () => {
    wrapper.find('p').simulate('click');
    expect(handleTouch).toHaveBeenCalledWith( demoTodos[0].id );
  });

  test('should debe mostrar el texto', () => {
    const p = wrapper.find("p");
    expect(p.text()).toBe(demoTodos[0].name);
  });

  test('should tener la clase completed si todo esta done', () => {
    const todo = demoTodos[0];
    todo.done = true;
    const wrapper = shallow(<TodoItem todo={ todo }/>);
    const p = wrapper.find("p");
    expect(p.hasClass('completed')).toBe(true);
  });
});
