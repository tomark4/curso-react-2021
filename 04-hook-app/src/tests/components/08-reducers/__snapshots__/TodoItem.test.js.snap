// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en todo item should mostrar el component 1`] = `
<li
  className="list-group-item"
>
  <div
    className="row"
  >
    <div
      className="col-md-8"
    >
      <p
        className=""
        onClick={[Function]}
      >
        lorem 1
      </p>
    </div>
    <div
      className="col-md-4 text-right"
    >
      <button
        className="btn btn-danger btn-sm"
        onClick={[Function]}
      >
        Borrar
      </button>
    </div>
  </div>
</li>
`;
